src/view/filter.rs

```diff
@@ -173,67 +217,67 @@ where
     }
 }
 
-impl<G, Ctx> LinkView for NodeFiltered<'_, G, NodeFilter<Ctx>, Ctx>
+impl<G, Ctx> LinkView for PortFiltered<'_, G, NodeFilter<Ctx>, PortFilter<Ctx>, Ctx>
 where
     G: LinkView,
 {
     type LinkEndpoint = G::LinkEndpoint;
 
-    type Neighbours<'a> = NodeFilteredIter<'a, G, Ctx, <G as LinkView>::Neighbours<'a>>
+    type Neighbours<'a> = PortFilteredIter<'a, G, Ctx, <G as LinkView>::Neighbours<'a>>
     where
         Self: 'a;
 
-    type NodeConnections<'a> = NodeFilteredIter<'a, G, Ctx, <G as LinkView>::NodeConnections<'a>>
+    type NodeConnections<'a> = PortFilteredIter<'a, G, Ctx, <G as LinkView>::NodeConnections<'a>>
     where
         Self: 'a;
 
-    type NodeLinks<'a> = NodeFilteredIter<'a, G, Ctx, <G as LinkView>::NodeLinks<'a>>
+    type NodeLinks<'a> = PortFilteredIter<'a, G, Ctx, <G as LinkView>::NodeLinks<'a>>
     where
         Self: 'a;
 
-    type PortLinks<'a> = NodeFilteredIter<'a, G, Ctx, <G as LinkView>::PortLinks<'a>>
+    type PortLinks<'a> = PortFilteredIter<'a, G, Ctx, <G as LinkView>::PortLinks<'a>>
     where
         Self: 'a;
 
     fn get_connections(&self, from: NodeIndex, to: NodeIndex) -> Self::NodeConnections<'_> {
```

Comment:
This should filter the links.
Same with `port_links`, `links`, `all_links`, `neighbours`, and `all_neighbours`.

Reply:
You're right, I missed `neighbours`, and `all_neighbours`. The other functions already filtered the links. Am I missing something?

Reply:
You're right. LGTM